lib/script.py

```diff
@@ -226,7 +226,15 @@ def generate_products(self):
         script = flags + commands
 
         for product in self.products:
-            script += product.generate()
+            items = product.generate()
+            for item in items:
+                if isinstance(item, list):
+                    for subitem in item:
+                        #TODO: What to do with this elements?
+                        #script += subitem
+                        continue
+                else:
+                    script += item
```

Comment:
Can you explain a bit more about what this is doing?

Reply:
Please add Lily to review this one. script is a string. items is a list containing strings and lists. On python 2.7 we can not append a list into a string, so we need to walk it, appending the strings on the array. 
The comment is to highlight the fact that I found no usage for the  nested lists... Not sure what the intended usage was.

Reply:
cc @millenomi

Reply:
IIRC: This path is triggered by the `+= [ … ]` command that adds the block runtime, which is invoked if libdispatch is not available. Can you just build libdispatch as part of your CF build instead?

Reply:
You can also fix that command to add single items instead of a list.

Reply:
Sorry, for context: this is the portion of build.py that is triggering that behavior:

```python
# This code is already in libdispatch so is only needed if libdispatch is
# NOT being used
if "LIBDISPATCH_SOURCE_DIR" not in Configuration.current.variables:
    sources += (['closure/data.c', 'closure/runtime.c'])
```

It'd be nice to fix this bit, but in general you want to set your `LIBDISPATCH_SOURCE_DIR` environment variable correctly.

Reply:
Applying LIBDISPATCH_SOURCE_DIR did not work for me. I just auggested a more pythonic way of applying the product commands... hopefully that is better.

Reply:
BTW, I'm compiling from within CoreFoundation, so the build.py I'm using is in CoreFoundation/build.py, that scripts do not have any references to LIBDISPATCH_SOURCE_DIR.